src/app/api/environment-variables.api.ts

```diff
@@ -0,0 +1,131 @@
+        accountName: string;
+        datasetName: string;
+        page: number;
+        perPage: number;
+    }): Observable<ListEnvVariablesQuery> {
+        return this.listEnvVariablesGQL.watch({ ...params }, noCacheFetchPolicy).valueChanges.pipe(
+            first(),
+            map((result: ApolloQueryResult<ListEnvVariablesQuery>) => {
+                return result.data;
+            }),
+        );
+    }
+
+    public saveEnvironmentVariable(params: {
+        accountId: string;
+        datasetId: string;
+        key: string;
+        value: string;
+        isSecret: boolean;
+    }): Observable<SaveEnvVariableMutation> {
+        return this.saveEnvVariableGQL
+            .mutate(
+                { ...params },
+                {
+                    update: (cache) => {
+                        const datasetKeyFragment = DatasetApi.generateDatasetKeyFragment(
+                            cache.identify(DatasetApi.generateAccountKeyFragment(params.accountId)),
+                            params.datasetId,
+                        );
+                        cache.evict({
+                            id: cache.identify(datasetKeyFragment),
+                            fieldName: "envVars",
+                        });
+                    },
+                },
+            )
+            .pipe(
+                first(),
+                map((result: MutationResult<SaveEnvVariableMutation>) => {
+                    /* istanbul ignore else */
+                    if (result.data) {
+                        return result.data;
+                    } else {
+                        throw new DatasetOperationError(result.errors ?? []);
+                    }
+                }),
+            );
+    }
+
+    public modifyEnvironmentVariable(params: {
+        datasetId: string;
+        id: string;
+        newValue: string;
+        isSecret: boolean;
+    }): Observable<ModifyEnvVariableMutation> {
+        return this.modifyEnvVariableGQL.mutate({ ...params }).pipe(
+            first(),
+            map((result: MutationResult<ModifyEnvVariableMutation>) => {
+                /* istanbul ignore else */
+                if (result.data) {
+                    return result.data;
+                } else {
+                    throw new DatasetOperationError(result.errors ?? []);
+                }
+            }),
+        );
+    }
+
+    public deleteEnvironmentVariable(params: {
+        datasetId: string;
+        datasetEnvVarId: string;
+    }): Observable<DeleteEnvVariableMutation> {
+        return this.deleteEnvVariableGQL.mutate({ ...params }).pipe(
+            first(),
+            map((result: MutationResult<DeleteEnvVariableMutation>) => {
+                /* istanbul ignore else */
+                if (result.data) {
+                    return result.data;
+                } else {
+                    throw new DatasetOperationError(result.errors ?? []);
+                }
+            }),
```

Comment:
What about delete - what's the effect from cache point of view?

Reply:
Fixed